framework/src/org/checkerframework/common/value/util/Range.java

```diff
@@ -767,6 +767,38 @@ public Range refineEqualTo(Range right) {
     }
 
     /**
+     * Refines this range to reflect that no value in it can be equal to a value in the given range.
+     * This is used for calculating the control-flow-refined result of the != operator. For example:
+     *
+     * <pre>
+     * <code>
+     *     {@literal @}IntRange(from = 0, to = 10) int a;
+     *     {@literal @}IntRange(from = 3, to = 15) int b;
+     *     ...
+     *     if (a != b) {
+     *         // range of <i>a</i> is now refined to [0, 2] because a value in range [3, 10]
```

Comment:
You're correct; this is my mistake. I was wondering why you'd implemented the other APIs and not that one, but I suppose I didn't think it through. I'll remove this part and add a test to ensure that it isn't reintroduced.

On an equality check we certainly can gain information, and your API gives that information correctly. But on an `!=` we shouldn't do a refinement.

Thanks for pointing this out!